subprojects/core-api/src/main/java/org/gradle/api/initialization/dsl/ScriptHandler.java

```diff
@@ -104,6 +105,17 @@ public interface ScriptHandler {
      */
     ConfigurationContainer getConfigurations();
 
+    /**
+     * Configures the configurations for the script. Executes the given closure against the {@link ConfigurationContainer} for
+     * this handler. The {@link ConfigurationContainer} is passed to the closure as the closure's delegate.
+     *
+     * @param configureClosure The closure used to configure the script configurations.
+     *
+     * @since 8.4
+     */
+    @Incubating
+    void configurations(Closure configureClosure);
```

Comment:
I tried to do that initially, but there was some weird error (I forget now, I wrote this a while ago). I suspected it has something to do with the fact that this is in the buildscript block and the auto-generation is potentially handled differently. 

I can go back and check what the original error was and see if I can dig deeper. I felt it was not _so_ important, since there are other methods in this class that use the `Closure` style

Reply:
When changed to an Action, the new groovy test fails with:
```
Caused by: org.gradle.internal.metaobject.AbstractDynamicObject$CustomMessageMissingMethodException: Could not find method foo() for arguments [build_29h8tzzwiu6pczslnnnlm1z9l$_run_closure1$_closure2$_closure3@60ccf498] on object of type org.gradle.api.internal.initialization.DefaultScriptHandler.
	at app//org.gradle.internal.metaobject.AbstractDynamicObject$CustomMissingMethodExecutionFailed.<init>(AbstractDynamicObject.java:190)
	at app//org.gradle.internal.metaobject.AbstractDynamicObject.methodMissingException(AbstractDynamicObject.java:184)
	at app//org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:80)
	at build_29h8tzzwiu6pczslnnnlm1z9l$_run_closure1$_closure2.doCall$original(/Users/jvandort/work/gradle/subprojects/dependency-management/build/tmp/teŝt files/ScriptDepen.Test/cufcu/build.gradle:4)
	at app//org.gradle.api.internal.initialization.DefaultScriptHandler.configurations(DefaultScriptHandler.java:133)
	at app//org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:525)
	at app//org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:216)
	at app//org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:61)
	at build_29h8tzzwiu6pczslnnnlm1z9l$_run_closure1.doCall$original(/Users/jvandort/work/gradle/subprojects/dependency-management/build/tmp/teŝt files/ScriptDepen.Test/cufcu/build.gradle:3)
	at app//org.gradle.util.internal.ClosureBackedAction.execute(ClosureBackedAction.java:73)
	at app//org.gradle.util.internal.ConfigureUtil.configureTarget(ConfigureUtil.java:155)
	at app//org.gradle.util.internal.ConfigureUtil.configure(ConfigureUtil.java:106)
	at app//org.gradle.api.internal.project.DefaultProject.buildscript(DefaultProject.java:1318)
	at app//org.gradle.api.internal.project.ProjectScript.buildscript(ProjectScript.java:47)
	at build_29h8tzzwiu6pczslnnnlm1z9l.run(/Users/jvandort/work/gradle/subprojects/dependency-management/build/tmp/teŝt files/ScriptDepen.Test/cufcu/build.gradle:2)
	at app//org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:91)
```

It looks like the action method is being called and no Closure method is called. Therefore, it tries to call a method named `foo` instead of going through `ConfigureUtil` to do all the NDOC dynamic groovy magic.

Reply:
Instantiating the `DefaultScriptHandler` with an instantiator instead of `new` fixes this.